saga/sagalogs/file_saga_log.go

```diff
@@ -0,0 +1,387 @@
+package sagalogs
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/scootdev/scoot/saga"
+)
+
+// Writes Saga Log to file system.  Not durable beyond machine failure
+// Sagas are stored in a directory.  Each saga has a corresponding directory
+// each saga directory contains a log file, and associated data files.
+
+// StartSaga Message
+// StartSaga \n
+// job data filename \n
+
+// StartTask Message
+// StartTask \n
+// taskId \n
+// taskData filename \n
+
+// EndTask Message
+// EndTask \n
+// taskId \n
+// taskData filename \n
+
+// AbortSaga Message
+// AbortSaga \n
+
+// StartCompTask Message
+// StartCompTask \n
+// taskId \n
+// taskData filename \n
+
+// EndCompTask Message
+// EndComptTask \n
+// taskId \n
+// taskData filename \n
+
+// EndSaga Message
+// EndSaga
+type fileSagaLog struct {
+	dirName string
+}
+
+// Creates a FileSagaLog with files stored at the specified directory
+// If the directory does not exist it will create it.
+func MakeFileSagaLog(dirName string) (*fileSagaLog, error) {
+
+	if _, err := os.Stat(dirName); err != nil {
+		if os.IsNotExist(err) {
+			os.Mkdir(dirName, os.ModePerm)
+		} else {
+			return nil, err
+		}
+	}
+
+	return &fileSagaLog{
+		dirName: dirName,
+	}, nil
+}
+
+// all files for a saga log are stored in a directory named
+// by the specified sagaId.
+func (log *fileSagaLog) getSagaDirectory(sagaId string) string {
+	return fmt.Sprintf("%v/%v", log.dirName, sagaId)
+}
+
+// Returns the name of the sagalog for the specified file.
+func (log *fileSagaLog) getSagaLogFileName(sagaId string) string {
+	return fmt.Sprintf("%v/%v", log.getSagaDirectory(sagaId), "log")
+}
+
+// Returns the name of the file to store task data in
```

Comment:
Storing the SagaMessage.Data? (data for saga start, end, abort as well as task messages)  if so then comment should read 'store saga message data' .